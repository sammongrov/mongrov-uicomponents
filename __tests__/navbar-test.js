// import { Platform } from 'react-native';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Icon, NavBar } from '@ui/components';
import renderer from 'react-test-renderer';

const mockMath = global.Math;
mockMath.random = () => 0.5;
global.Math = mockMath;

jest.mock('Platform', () => {
  const Platform = require.requireActual('Platform');
  Platform.OS = 'android';
  return Platform;
});

it('navbar title renders correctly', () => {
  const tree = renderer
    .create(
      <NavBar
        titleComponent={
          <TouchableOpacity>
            <Icon name="chevron-left" color="red" size={24} />
          </TouchableOpacity>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('chat navbar renders correctly', () => {
  const tree = renderer.create(<NavBar chatTitle="John" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('navbar title renders correctly', () => {
  const tree = renderer.create(<NavBar titleText="John" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('type props returns null correctly', () => {
  const inst = renderer.create(<NavBar chatInfo="online" type="d" />);
  expect(inst.getInstance()._renderChatInfo('d')).toBeTruthy();
  expect(inst.getInstance()._renderChatInfo()).toBeNull();
});

it('type props returns null correctly', () => {
  const inst = renderer.create(<NavBar chatSubTitle="group" type="d" />);
  expect(inst.getInstance()._renderChatInfo('d')).toBeFalsy();
  expect(inst.getInstance()._renderChatInfo()).toBeNull();
});
