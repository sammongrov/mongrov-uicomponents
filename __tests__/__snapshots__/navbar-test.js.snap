// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat navbar renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      null,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 18,
            "height": 36,
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "OpenSans-Regular",
            },
            Array [
              undefined,
            ],
          ]
        }
      >
        John
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`navbar title renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      null,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ÔÖÅ
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
</View>
`;

exports[`navbar title renders correctly 2`] = `
<View
  style={
    Array [
      undefined,
      null,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        undefined,
        undefined,
        null,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Regular",
          },
          Array [
            undefined,
            null,
          ],
        ]
      }
    >
      John
    </Text>
  </View>
  <View
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
